### PR TITLE
dune: add page

### PR DESCRIPTION
**Pull request by dmbaturin**
[Dune](https://dune.build) is the most popular build system for the OCaml programming language now, used by many (if not most) packages from the OPAM repository.

- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

 - **Version of the command being documented (if known):** 2.9, but commands used in examples existed for a long time and the interface is stable

### CODE REVIEW

File: pages/common/dune.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# dune
+
+> A build system for [OCaml](https://ocaml.org) programs.
```
Comment (CleanMachine1) [inline pages/common/dune.md:3]:
```suggestion
> A build system for OCaml programs.
```
This is what the more information link is for.

File: pages/common/dune.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+- Clean up the workspace:
+
+`dune clean`
+
+- Run tests:
+
+`dune runtest`
+
+- Load compiled libraries in the [utop](https://github.com/ocaml-community/utop) REPL:
```
Comment (CleanMachine1) [inline pages/common/dune.md:18]: Could you explain what this does.

**Review by CleanMachine1 — CHANGES_REQUESTED** [review]

File: pages/common/dune.md (hunk 1 above)
Comment (dmbaturin) [reply pages/common/dune.md]: I've removed the link. However, those were different links: the first link is to the OCaml programming language website, the "more information" one was to the website of the Dune build tool, which is a third-party project. I believe for less-than-mainstream tools a link to the language would not be useless.

File: pages/common/dune.md (hunk 2 above)
Comment (dmbaturin) [reply pages/common/dune.md]: I reworded the passage. Let me know if it's still not clear.

File: pages/common/dune.md (hunk 1 above)
Comment (marchersimon) [reply pages/common/dune.md]: Most clients (and terminals) don't support markdown syntax for links. If the link is important we could add in it parentheses.

**Comment by marchersimon** [comment]
Please try to avoid force-pushing. It makes reviewing a bit harder and we squash all commits before merging anyways.

File: pages/common/dune.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+- Build all targets:
+
+`dune build`
+
+- Clean up the workspace:
+
+`dune clean`
+
+- Run tests:
```
Comment (marchersimon) [inline pages/common/dune.md:14]:
```suggestion
- Run all tests:
```

File: pages/common/dune.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+`dune runtest`
+
+- Start the [utop](https://github.com/ocaml-community/utop) REPL with compiled modules automatically loaded into it, to remove the need to load them by hand:
```
Comment (marchersimon) [inline pages/common/dune.md:18]:
```suggestion
- Start the utop REPL with compiled modules automatically loaded into it, to remove the need to load them by hand:
```

**Review by marchersimon — APPROVED** [review]
Thanks @dmbaturin

**Review by navarroaxel — APPROVED** [review]

**Pull request merged**